dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ApacheHttpClientBlockingChannel.java

```diff
@@ -100,8 +102,9 @@ final class ApacheHttpClientBlockingChannel implements BlockingChannel {
     public Response execute(Endpoint endpoint, Request request) throws IOException {
         // Create base request given the URL
         URL target = baseUrl.render(endpoint, request);
+        URI targetUri = Uris.tryParse(target.toString()).uriOrThrow();
```

Comment:
Unlike the base URIs, this value will have much higher cardinality any time an endpoint includes path parameters. I'm a little bit uneasy about caching.

We could bypass the URL and URI types entirely, setting specific elements directly on the request builder instead. That would require expanding the BaseUrl API a bit, but may produce better results. What do you think?

Reply:
good call, updated to just use the URL protocol, authority, and file which avoids some of the more expensive `new URI(String)` parsing and not flooding the URI parsing cache with high cardinality URIs

Reply:
split this portion out to a separate PR https://github.com/palantir/dialogue/pull/2437 as this is portion is the biggest win by avoiding a `URI` parse per request.